### PR TITLE
feat(storage-resize-images): put failed resizes in a separate directory

### PR DESCRIPTION
**Pull request by cabljac**
fixes: https://github.com/firebase/extensions/issues/563

Note
If the extension is configured to generate multiple sizes, the image will be copied to the failed folder even if only one size fails

### CODE REVIEW

**Comment by cabljac** [comment]
needs specific tests adding

File: storage-resize-images/functions/__tests__/e2e.test.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    // wait for file to be uploaded to storage:
+    expect(await waitForFile(successFilePath)).toBe(true);
+
+    const failureFilePath = `${process.env.FAILED_IMAGES_PATH}/not-an-image.jpeg`;
+
+    expect(await waitForFile(failureFilePath)).toBe(true);
+  });
+});
+
+const waitForFile = async (
```
Comment (dackers86) [inline storage-resize-images/functions/__tests__/e2e.test.ts:50]: Potentially move to a util file or folder for reusability?

File: storage-resize-images/functions/__tests__/e2e.test.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    }
+    storage = admin.storage();
+    await storage.bucket().upload(__dirname + "/not-an-image.jpeg", {});
+    await storage.bucket().upload(__dirname + "/test-image.jpeg", {});
+  });
+  afterEach(async () => {
+    // await storage.bucket().deleteFiles(process.env.RESIZED_IMAGES_PATH);
+  });
+
+  test("should resize image successfully, and copy failed image to failed directory", async () => {
```
[the remark this answers is not in the document]
Comment (cabljac) [reply storage-resize-images/functions/__tests__/e2e.test.ts:39]: lol no, will update!

File: storage-resize-images/functions/__tests__/e2e.test.ts (hunk 1 above)
Comment (cabljac) [reply storage-resize-images/functions/__tests__/e2e.test.ts:50]: yup!

**Review by dackers86 — APPROVED** [review]

**Pull request merged**